src/server/protocol.ts

```diff
@@ -1829,6 +1830,38 @@ namespace ts.server.protocol {
         closedFiles?: string[];
     }
 
+    export interface UpdateFileSystemRequest extends Request {
+        command: CommandTypes.UpdateFileSystem;
+        arguments: UpdateFileSystemRequestArgs;
+    }
+
+    /**
+     * Arguments to UpdateFileSystemRequest. Tracks changes to non-opened files.
+     */
+    export interface UpdateFileSystemRequestArgs {
+        /** For now, only 'memfs', initially for exclusive in-memory operation, but it could be other in-memory names later */
+        fileSystem: string;
+        /** List of newly created or newly available files. */
+        created: FileSystemRequestArgs[];

```

Comment:
Is it an error to create a file that already exists?  If not, what's the point of having a separate `updated` property?  Do those files provide diffs?

Reply:
No, and at least for the first version, there's no plan for diffs. I think I'll try simplifying the protocol to have `files` and `deleted`; then we can add `diffs` later if that's worth having.